packages/typeahead/components/Typeahead.tsx

```diff
@@ -0,0 +1,252 @@
+import * as React from "react";
+import Downshift from "downshift";
+import Dropdownable from "../../dropdownable/components/Dropdownable";
+import DropdownMenu from "../../dropdowMenu/components/DropdownMenu";
+import DropdownMenuItem from "../../dropdowMenu/components/DropdownMenuItem";
+import { margin } from "../../shared/styles/styleUtils";
+import resizeEventManager from "../../utilities/resizeEventManager";
+
+export interface Item {
+  value: string;
+  label: React.ReactNode;
+}
+
+export interface TypeaheadProps {
+  /**
+   * items to render into the menu
+   */
+  items: Item[];
+  /**
+   * the text input element
+   */
+  textField: React.ReactElement<any>;
+  /**
+   * what is displayed when there are no items
+   */
+  menuEmptyState?: React.ReactElement<any>;
+  /**
+   * maximum height the menu can grow to
+   */
+  menuMaxHeight?: number;
+  /**
+   * whether more than one item can be selected
+   */
+  multiSelect?: boolean;
+  /**
+   * callback for when an item is selected
+   */
+  onSelect?: (selectedItems: string[]) => void;
+  /**
+   * which DOM node the dropdown menu will attach to
+   */
+  overlayRoot?: HTMLElement;
```

Comment:
I had to add this to support usage inside of modals in dcos-ui. If it portals to `document.body`, it will be covered by a modal because modals in dcos-ui have `z-index: 999`.